packages/flutter/lib/src/material/chip.dart

```diff
@@ -2188,29 +2188,40 @@ class _RenderChip extends RenderBox {
     // Now that we know the label height and the width of the icons, we can
     // determine how much to shrink the width constraints for the "real" layout.
     if (constraints.maxWidth.isFinite) {
+      final double maxWidth = math.max(
+        0.0,
+        constraints.maxWidth
+        - iconSizes
+        - theme.labelPadding.horizontal
```

Comment:
Is the labelPadding applied unconditionally, or only if there's an avatar and a deleteIcon (in that case would it be applied twice?

Reply:
It should be applied unconditionally according to https://material.io/design/components/chips.html#specs